### PR TITLE
Migrate to ember

### PR DESCRIPTION
**Pull request by armanbilge**
Now that blaze is schismed, the dependency is more of a burden here. This is especially a problem on main, because it means the jdk-http-client is blocked from updating to the latest http4s milestone until blaze itself has updated and released, which may not be straightforward.

### CODE REVIEW

**Comment by armanbilge** [comment]
Actually, I think the tests are (hopefully) succeeding, just ember is hanging on cancellation 🤔

**Comment by armanbilge** [comment]
- https://github.com/http4s/http4s/issues/6420

**Comment by amesgen** [comment]
Switching to ember is great (it just didn't have a WS server back then :laughing:). Thanks for creating the issue, I am also taking a look at that cancellation bug.

**Comment by armanbilge** [comment]
blaze released against 1.0.0-M33, so the jdk-http-client is unblocked for now.

If the upstream bugs aren't resolved by M34, I think we should consider switching to the scaffold server (which is based on netty) and Java-WebSocket for tests to become independent of blaze.

**Comment by amesgen** [comment]
> blaze released against 1.0.0-M33, so the jdk-http-client is unblocked for now.

Opened #648 for that

**Comment by armanbilge** [comment]
Lol, we've discovered another ember bug 😂 
https://github.com/http4s/http4s/blob/edaf32c16ff269eaafaeda396f0441258a8f0224/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala#L122

**Comment by armanbilge** [comment]
Alright! I think some fresh eyes did the trick 😅

File: build.sbt [hunk 1]
```diff
@@ -62,7 +62,7 @@ val coreDeps = Seq(
   "org.scodec" %% "scodec-bits" % scodecV,
   "org.typelevel" %% "vault" % vaultV,
   "org.typelevel" %% "case-insensitive" % caseInsensitiveV
-) ++ (blazeServer ++ Seq(
+) ++ (emberServer ++ Seq(
   "org.http4s" %% "http4s-client-testkit" % http4sV,
   "org.java-websocket" % "Java-WebSocket" % javaWebsocketV,
```
[the remark this answers is not in the document]
Comment (amesgen) [reply build.sbt:70]: Do we want to try to do this as part of this PR? I would be fine either way.

**Review by amesgen — APPROVED** [review]
LGTM, only one comment about the scope of this PR :+1:

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (armanbilge) [reply build.sbt:70]: Oh I completely forgot about this! I do want to make this go away, it should be easy. Let me give a try.

File: core/src/test/scala/org/http4s/jdkhttpclient/JdkWSClientSpec.scala [hunk 2]
```diff
@@ -135,44 +130,37 @@ class JdkWSClientSpec extends CatsEffectSuite {
...
-              conn.send(WSFrame.Text("hey blaze"))
+      server = EmberServerBuilder
+        .default[IO]
+        .withPort(port"0")
+        .withHttpWebSocketApp { wsb =>
+          HttpRoutes
+            .of[IO] { case GET -> Root =>
+              wsb
+                .withOnClose(ref.update(_ :+ closeFrame))
+                .build(_.evalTap(f => ref.update(_ :+ f)))
```
Comment (armanbilge) [inline core/src/test/scala/org/http4s/jdkhttpclient/JdkWSClientSpec.scala:144]: This `onClose` is a bit awkward but whatever, just a test 😅

**Review by amesgen — APPROVED** [review]
Thanks, this is a great cleanup of the test suite!

**Pull request merged**